tsl/src/compression/compression.c

```diff
@@ -2107,6 +2131,7 @@ decompress_batches_for_insert(ChunkInsertState *cis, Chunk *chunk, TupleTableSlo
 	table_endscan(heapScan);
 
 	ts_catalog_close_indexes(decompressor.indexstate);
+	FreeExecutorState(decompressor.estate);
```

Comment:
Out of scope for this PR: we should create a decompressor finisher/closer function, seems like we have some management at the end of its lifecycle.